### PR TITLE
Added single, singleOrNil, singleOrDefault

### PR DESCRIPTION
**Pull request by intrepidmatt**

### CODE REVIEW

File: SINQ/SINQ.swift [hunk 1]
```diff
@@ -618,7 +618,41 @@ public class SinqSequence<T>: SequenceType {
         return selectMany({ (x, _) in selector(x) }, result: { $0 })
     }
     
-    // TODO: single
+    public func single() -> T {
+        return self.singleOrNil()!
+    }
+    
+    public func singleOrNil() -> T? {
+        var gen = self.generate()
```
Comment (slazyk) [inline SINQ/SINQ.swift]: why not switch over (gen.next(), gen.next()) ?

**Comment by slazyk** [comment]
Can you add it also to the README.md?

File: SINQ/SINQ.swift (hunk 1 above)
Comment (intrepidmatt) [inline SINQ/SINQ.swift]: Good point, will refactor and push

**Comment by intrepidmatt** [comment]
Added readme and refactored singleOrNil()

**Pull request merged**